### ISSUE TITLE
BUG: (EthereumJSONRPC.DecodeError) Failed to decode Ethereum JSONRPC response

### ISSUE THREAD

**Issue by youngqqcn** [blockscout/blockscout#6010]
*Describe your issue here.*

### Environment

* Elixir & Erlang/OTP versions (`elixir -version`): 
* Operating System:
* Blockscout Version/branch:  tag4.1.7-beta

### Steps to reproduce

*Tell us how to reproduce this issue.  ❤️ if you can push up a branch to your fork with a regression test we can run to reproduce locally.*

### Expected behaviour

*Tell us what should happen.*

### Actual behaviour

*Tell us what happens instead.*

```

    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/http.ex:117: EthereumJSONRPC.HTTP.decode_json/1
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/http.ex:69: EthereumJSONRPC.HTTP.chunked_json_rpc/3
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/request_coordinator.ex:86: anonymous fn/3 in EthereumJSONRPC.RequestCoordinator.perform/4
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/request_coordinator.ex:106: EthereumJSONRPC.RequestCoordinator.trace_request/2
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc.ex:483: EthereumJSONRPC.fetch_blocks_by_params/3
    (indexer 0.1.0) lib/indexer/block/fetcher.ex:131: Indexer.Block.Fetcher.fetch_and_import_range/2
    (indexer 0.1.0) lib/indexer/block/catchup/fetcher.ex:209: Indexer.Block.Catchup.Fetcher.fetch_and_import_range_from_sequence/3
    (elixir 1.13.3) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
    (elixir 1.13.3) lib/task/supervised.ex:34: Task.Supervised.reply/4
    (stdlib 3.17.2) proc_lib.erl:226: :proc_lib.init_p_do_apply/3


Retrying.
2022-08-30T18:33:03.455 application=indexer fetcher=block_catchup first_block_number=117729 last_block_number=117720 [error] ** (EthereumJSONRPC.DecodeError) Failed to decode Ethereum JSONRPC response:

  request:

    url: http://172.16.100.104:8545

    body: [{"id":0,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CBE1",true]},{"id":1,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CBE0",true]},{"id":2,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CBDF",true]},{"id":3,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CBDE",true]},{"id":4,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CBDD",true]},{"id":5,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CBDC",true]},{"id":6,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CBDB",true]},{"id":7,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CBDA",true]},{"id":8,"jsonrpc":"2.0","method":"e

  response:

    status code: 200

    body: 

    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/http.ex:117: EthereumJSONRPC.HTTP.decode_json/1
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/http.ex:69: EthereumJSONRPC.HTTP.chunked_json_rpc/3
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/request_coordinator.ex:86: anonymous fn/3 in EthereumJSONRPC.RequestCoordinator.perform/4
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/request_coordinator.ex:106: EthereumJSONRPC.RequestCoordinator.trace_request/2
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc.ex:483: EthereumJSONRPC.fetch_blocks_by_params/3
    (indexer 0.1.0) lib/indexer/block/fetcher.ex:131: Indexer.Block.Fetcher.fetch_and_import_range/2
    (indexer 0.1.0) lib/indexer/block/catchup/fetcher.ex:209: Indexer.Block.Catchup.Fetcher.fetch_and_import_range_from_sequence/3
    (elixir 1.13.3) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
    (elixir 1.13.3) lib/task/supervised.ex:34: Task.Supervised.reply/4
    (stdlib 3.17.2) proc_lib.erl:226: :proc_lib.init_p_do_apply/3


Retrying.
2022-08-30T18:33:04.746 application=indexer fetcher=block_catchup first_block_number=117569 last_block_number=117560 [error] ** (EthereumJSONRPC.DecodeError) Failed to decode Ethereum JSONRPC response:

  request:

    url: http://172.16.100.104:8545

    body: [{"id":0,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CB41",true]},{"id":1,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CB40",true]},{"id":2,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CB3F",true]},{"id":3,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CB3E",true]},{"id":4,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CB3D",true]},{"id":5,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CB3C",true]},{"id":6,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CB3B",true]},{"id":7,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CB3A",true]},{"id":8,"jsonrpc":"2.0","method":"e

  response:

```

**Comment by youngqqcn** [comment]
```
curl -H "Content-Type: application/json" -X POST --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CBE0", true],"id":1}' 172.16.100.104:8545 -s | jq | more


{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "baseFeePerGas": "0x0",
    "difficulty": "0x2",
    "extraData": "0xd883010000846765746888676f312e31382e34856c696e757800000000000000f9d7fec7f33856a3be172e87d4942cffea05ca718d663ae6f0e25fa1ec8a429e2381e97aeb8b7ac458d86b18d8f2bbc464e94874c2526d075a86f
7d6896cc28b01",
    "gasLimit": "0x2faf080",
    "gasUsed": "0xe75e20",
    "hash": "0x6ac18d4a1dd0ef34a8a7f6a81d2181296d067ad352e4965acda00866a5643b91",
    "logsBloom": "0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "miner": "0x0d80b670b2d44017468ccc75787ea017678c6f72",
    "mixHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "nonce": "0x0000000000000000",
    "number": "0x1cbe0",
    "parentHash": "0x0c8547bc7935f5c4d7efb674699eb109cba225e2065084628d0407d9a207dfb0",
    "receiptsRoot": "0xfeb4890a7150729877d32d1a319e83336c64b2328a6234c5a92a22dab41fcfe3",
    "sha3Uncles": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
    "size": "0x943ea",
    "stateRoot": "0x7c637c18ac9effbec188b396a29c80cf34ea33d034876631b299861092ef945c",
    "timestamp": "0x630de6d1",
    "totalDifficulty": "0x39790",
    "transactions": [
      {
        "blockHash": "0x6ac18d4a1dd0ef34a8a7f6a81d2181296d067ad352e4965acda00866a5643b91",
        "blockNumber": "0x1cbe0",
        "from": "0xeb87c31698cb33f0f1bfc7728cb3db4a28c3d935",
        "gas": "0x13880",
        "gasPrice": "0x2540be400",
        "hash": "0xc266252433b01d11867387ad7b36f06123d590a11cc0cfe0931f5b2198dee9ef",
      
        ...

     {
        "blockHash": "0x6ac18d4a1dd0ef34a8a7f6a81d2181296d067ad352e4965acda00866a5643b91",
        "blockNumber": "0x1cbe0",
        "from": "0x584ded0aaea7962e55aa412b2077e581e9f6bf9c",
        "gas": "0x13880",
        "gasPrice": "0x2540be400",
        "hash": "0xef4810e129952863de9a689bd09fd769da79b30db23644a909705d6847060c06",
        "input": "0x1f7fdffa000000000000000000000000b79c3c236f72fb3970ffee36703c98a9022a23fb0000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000012000000000000000000000000000000000000000000000000000000000000001c0000000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000027740000000000000000000000000000000000000000000000000000000000002775000000000000000000000000000000000000000000000000000000000000277600000000000000000000000000000000000000000000000000000000000027770000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000003b9aca00000000000000000000000000000000000000000000000000000000e8d4a51000000000000000000000000000000000000000000000000000000000003b9aca0000000000000000000000000000000000000000000000000000038d7ea4c6800000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
        "nonce": "0x14eb50",
        "to": "0x992589bcd72aedf6cf653917022ab15573ef5daf",
        "transactionIndex": "0x257",
        "value": "0x0",
        "type": "0x0",
        "v": "0x601e",
        "r": "0x57f7930c739c27973c8fbf586855ebb6e9bf5dd6f557747e06a29652ced61baa",
        "s": "0x35bf7aa4a9ff7e433045f5dce96abc0455fee3b393ce0c8226fbd042c5201d"
      }
    ],
    "transactionsRoot": "0xd631289e200a76aa269a1fddb951b93cbfe1d70262f180ce8b884686205a0b59",
    "uncles": []
  }
}

```

**Comment by youngqqcn** [comment]
```
response:

    status code: 200

    body: 

    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/http.ex:117: EthereumJSONRPC.HTTP.decode_json/1
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/http.ex:69: EthereumJSONRPC.HTTP.chunked_json_rpc/3
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/request_coordinator.ex:86: anonymous fn/3 in EthereumJSONRPC.RequestCoordinator.perform/4
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/request_coordinator.ex:106: EthereumJSONRPC.RequestCoordinator.trace_request/2
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc.ex:483: EthereumJSONRPC.fetch_blocks_by_params/3
    (indexer 0.1.0) lib/indexer/block/fetcher.ex:131: Indexer.Block.Fetcher.fetch_and_import_range/2
    (indexer 0.1.0) lib/indexer/block/catchup/fetcher.ex:209: Indexer.Block.Catchup.Fetcher.fetch_and_import_range_from_sequence/3
    (elixir 1.13.3) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
    (elixir 1.13.3) lib/task/supervised.ex:34: Task.Supervised.reply/4
    (stdlib 3.17.2) proc_lib.erl:226: :proc_lib.init_p_do_apply/3


Retrying.
2022-08-31T12:14:20.086 application=indexer fetcher=block_catchup first_block_number=137748 last_block_number=137739 [error] ** (EthereumJSONRPC.DecodeError) Failed to decode Ethereum JSONRPC response:

  request:

    url: http://172.16.100.104:8545

    body: [{"id":0,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x21A14",true]},{"id":1,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x21A13",true]},{"id":2,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x21A12",true]},{"id":3,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x21A11",true]},{"id":4,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x21A10",true]},{"id":5,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x21A0F",true]},{"id":6,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x21A0E",true]},{"id":7,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x21A0D",true]},{"id":8,"jsonrpc":"2.0","method":"e

  response:

    status code: 200

    body: 

    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/http.ex:117: EthereumJSONRPC.HTTP.decode_json/1
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/http.ex:69: EthereumJSONRPC.HTTP.chunked_json_rpc/3
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/request_coordinator.ex:86: anonymous fn/3 in EthereumJSONRPC.RequestCoordinator.perform/4
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc/request_coordinator.ex:106: EthereumJSONRPC.RequestCoordinator.trace_request/2
    (ethereum_jsonrpc 0.1.0) lib/ethereum_jsonrpc.ex:483: EthereumJSONRPC.fetch_blocks_by_params/3
    (indexer 0.1.0) lib/indexer/block/fetcher.ex:131: Indexer.Block.Fetcher.fetch_and_import_range/2
    (indexer 0.1.0) lib/indexer/block/catchup/fetcher.ex:209: Indexer.Block.Catchup.Fetcher.fetch_and_import_range_from_sequence/3
    (elixir 1.13.3) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
    (elixir 1.13.3) lib/task/supervised.ex:34: Task.Supervised.reply/4
    (stdlib 3.17.2) proc_lib.erl:226: :proc_lib.init_p_do_apply/3


Retrying.

```

**Comment by youngqqcn** [comment]
conbine 2 request in a request body :
```
curl -H "Content-Type: application/json" -X POST --data '[{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CBE0", true],"id":1},{"id":0,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x21A14",true]}]' 172.16.100.104:8545 -s | jq

response is ok ， but response is too large
```


count response by wc
```
curl -H "Content-Type: application/json" -X POST --data '[{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1CBE0", true],"id":1},{"id":0,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x21A14",true]}]' 172.16.100.104:8545 -s | wc -c

2847703
```

**Comment by youngqqcn** [comment]
```
2022-08-31T17:12:07.000 application=indexer fetcher=block_catchup first_block_number=136443 last_block_number=136441 [error] ** (Postgrex.Error) ERROR 40P01 (deadlock_detected) deadlock detected

    hint: See server log for query details.

Process 1381 waits for ShareLock on transaction 4494396; blocked by process 1428.
Process 1428 waits for ShareLock on transaction 4494402; blocked by process 1381.
    (ecto_sql 3.7.1) lib/ecto/adapters/sql.ex:760: Ecto.Adapters.SQL.raise_sql_call_error/1
    (ecto_sql 3.7.1) lib/ecto/adapters/sql.ex:667: Ecto.Adapters.SQL.insert_all/9
    (ecto 3.7.1) lib/ecto/repo/schema.ex:58: Ecto.Repo.Schema.do_insert_all/7
    (explorer 0.0.1) lib/explorer/repo.ex:61: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
    (elixir 1.13.3) lib/enum.ex:2396: Enum."-reduce/3-lists^foldl/2-0-"/3
    (explorer 0.0.1) lib/explorer/chain/import.ex:299: Explorer.Chain.Import.insert_changes_list/3
    (ecto 3.7.1) lib/ecto/multi.ex:716: Ecto.Multi.apply_operation/5
    (elixir 1.13.3) lib/enum.ex:2396: Enum."-reduce/3-lists^foldl/2-0-"/3
    (ecto 3.7.1) lib/ecto/multi.ex:690: anonymous fn/5 in Ecto.Multi.apply_operations/5
    (ecto_sql 3.7.1) lib/ecto/adapters/sql.ex:1021: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
    (db_connection 2.4.0) lib/db_connection.ex:1512: DBConnection.run_transaction/4
    (ecto 3.7.1) lib/ecto/repo/transaction.ex:20: Ecto.Repo.Transaction.transaction/4
    (stdlib 3.17.2) timer.erl:197: :timer.tc/3
    (explorer 0.0.1) lib/explorer/repo.ex:37: anonymous fn/2 in Explorer.Repo.logged_transaction/2
    (explorer 0.0.1) lib/explorer/logger.ex:14: Explorer.Logger.metadata/2
    (explorer 0.0.1) lib/explorer/chain/import.ex:330: anonymous fn/3 in Explorer.Chain.Import.import_transactions/2
    (elixir 1.13.3) lib/enum.ex:4475: Enumerable.List.reduce/3
    (elixir 1.13.3) lib/enum.ex:2442: Enum.reduce_while/3
    (explorer 0.0.1) lib/explorer/chain/import.ex:329: Explorer.Chain.Import.import_transactions/2
    (explorer 0.0.1) lib/explorer/logger.ex:14: Explorer.Logger.metadata/2


```

**Comment by vbaranov** [comment]
Please check against the latest version of Blockscout. If the error persists, please create a new issue.
